src/AgentNodePlugin.groovy

```diff
@@ -36,10 +44,14 @@ public class AgentNodePlugin implements TerraformValidateStagePlugin, TerraformE
 
     public Closure addAgent() {
         return { closure ->
-            if (dockerImage) {
+            if (this.dockerImage && this.withDockerfile == false) {
                 docker.image(this.dockerImage).inside(this.dockerOptions) {
                     closure()
                 }
+            } else if (this.dockerImage && this.withDockerfile == true) {
```

Comment:
It's minor, but `withDockerfile == true`, `withDockerfile == false` feels wonky.  

You could collapse this logic into:
```
if (dockerImage) {
    if (withDockerfile) {
        docker.build(...)
    } else {
        docker.image(...)
    }
} else {
    ....
}
```
Or at the very least, not compare against the literals:
```
if (this.dockerImage && !this.withDockerfile) {
    ...
} else if (this.dockerImage && this.withDockerfile) {
    ...
}
```

Reply:
Ah, I think this is why it feels wonky to me - this is  in the Sun Coding Standard - https://github.com/checkstyle/checkstyle/blob/master/src/main/resources/sun_checks.xml#L171

Reply:
I can change it either way. I tend to write literal checks so you know exactly what the code is looking for